code-challenges.js

```diff
@@ -7,6 +7,50 @@ var collections = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
+//create a function that takes in an array
+// const colorFilterShuffle = (array) => {
+// //shuffle all the items in the array
+//     for (let i = array.length - 1; i > 0; i--) {
+//         let j = Math.floor(Math.random() * (i + 1));
+//         [array[i], array[j]] = [array[j], array[i]];}
+//     //filter the original array, take in the value and the index
+//     return array.filter((value, index) => { 
+//     // in order to return the array with the first item removed
+//     return index > 0
+//     })
+// }
+
+// console.log(colorFilterShuffle(collections))
+
+//I was able to get the output to return the original array, shuffled, and without the first value, but I wasn't sure how to make this loop continuously so it always removes the first value until the array is empty.
+//I thought about putting it into a for loop so it would iterate through the length of the array and remove a value each time but it didn't seem to work.
+//I think the answer might be to push the values into a new array? But then Im not sure how to make it so the function takes in the new version of the array each time.
+
+//create a function that takes in an array
+const colorFilterShuffle2 = (array) => {
+    let newArr = []
+    for(let i=0; i<array.length; i++){
+        if(array != null){
+    //shuffle all the items in the array
+        for (let i = array.length - 1; i > 0; i--) {
+            let j = Math.floor(Math.random() * (i + 1));
+            [array[i], array[j]] = [array[j], array[i]];}
+        //filter the original array, take in the value and the index
+        return array.filter((value, index) => { 
+        // in order to return the array with the first item removed
+        return index > 0
+        })
+    }
+        else {
+            return "The array is empty"
+        }
+}
+}
```

Comment:
so this solution totally shuffles! but im not seeing any of the values being taken out of the array. but I feel like you're super close! you will also want to avoid nested for loops out in the world but they certainly are a cool concept ;)